heron/spi/src/java/com/twitter/heron/spi/packing/PackingPlan.java

```diff
@@ -253,6 +279,13 @@ public int hashCode() {
       return result;
     }
 
+    /**
+     * Returns a {@link ContainerPlan} with updated scheduledResource
+     */
+    public ContainerPlan cloneWithScheduledResource(Resource resource) {
```

Comment:
`clone` methods are public. Could you please help me understand why this should be private?